### PR TITLE
EOS-15411: resolv.conf update missed during unboxing

### PR DESCRIPTION
**Pull request by ypise**
Signed-off-by: Yashodhan Pise <yashodhan.pise@seagate.com>

### CODE REVIEW

File: cli/src/factory_ops/unboxing/network/public_mgmt [hunk 1]
```diff
@@ -98,6 +100,8 @@ function update_cluster_pillar_for_mgmt_ips {
         provisioner pillar_set --local cluster/srvnode-1/network/mgmt_nw/public_ip_addr \"${ip_1}\"
         provisioner pillar_set --local cluster/srvnode-1/network/mgmt_nw/gateway \"${gw_ip}\"
         provisioner pillar_set --local cluster/srvnode-1/network/mgmt_nw/netmask \"${netmask}\"
+        provisioner pillar_set --local cluster/srvnode-1/network/mgmt_nw/netmask \"${search_domain}\"
```
Comment (andkononykhin2) [inline cli/src/factory_ops/unboxing/network/public_mgmt:103]: two things here:
- a typo `netmask`
- search domains value should be a properly constructed json list (even if it's a one item list)

File: cli/src/factory_ops/unboxing/network/public_mgmt [hunk 2]
```diff
@@ -109,6 +113,8 @@ function update_cluster_pillar_for_mgmt_ips {
         provisioner pillar_set --local cluster/srvnode-2/network/mgmt_nw/public_ip_addr \"${ip_2}\"
         provisioner pillar_set --local cluster/srvnode-2/network/mgmt_nw/gateway \"${gw_ip}\"
         provisioner pillar_set --local cluster/srvnode-2/network/mgmt_nw/netmask \"${netmask}\"
+        provisioner pillar_set --local cluster/srvnode-2/network/mgmt_nw/netmask \"${search_domain}\"
+        provisioner pillar_set --local cluster/srvnode-2/network/mgmt_nw/netmask \"${dns_server}\"
```
Comment (andkononykhin2) [inline cli/src/factory_ops/unboxing/network/public_mgmt:117]: the same as above:
- a typo (`netmask` is copied for both)
- json list string is expected here

**Review by andkononykhin2 — CHANGES_REQUESTED** [review]
I guess there are some typos and places where json strings are not respected.

File: cli/src/factory_ops/unboxing/network/public_mgmt (hunk 2 above)
Comment (ypise) [reply cli/src/factory_ops/unboxing/network/public_mgmt:117]: Addressed

File: cli/src/factory_ops/unboxing/network/public_mgmt (hunk 1 above)
Comment (ypise) [reply cli/src/factory_ops/unboxing/network/public_mgmt:103]: Addressed

File: cli/src/factory_ops/unboxing/config_update [hunk 3]
```diff
@@ -335,35 +335,64 @@ function update_pillar_from_local {
         provisioner pillar_set cluster/srvnode-1/network/mgmt_nw/netmask \"${srv1_public_netmask}\"
     fi
 
+    srv1_search_domains=$(salt-call --local pillar.get cluster:srvnode-1:network:mgmt_nw:search_domains --output=newline_values_only)
+    if [[ "${srv1_search_domains}" == "None" ]]; then
+        provisioner pillar_set cluster/srvnode-1/network/mgmt_nw/search_domains PRVSNR_UNDEFINED
+    else
+        provisioner pillar_set cluster/srvnode-1/network/mgmt_nw/search_domains \"${srv1_search_domains}\"
```
[the remark this answers is not in the document]
Comment (ypise) [reply cli/src/factory_ops/unboxing/config_update]: Corrected

File: cli/src/factory_ops/unboxing/config_update [hunk 4]
```diff
@@ -335,35 +335,64 @@ function update_pillar_from_local {
...
+    fi
+
+    srv1_public_dns_servers=$(salt-call --local pillar.get cluster:srvnode-1:network:mgmt_nw:dns_servers --output=newline_values_only)
+    if [[ "${srv1_public_dns_servers}" == "None" ]]; then
+        provisioner pillar_set cluster/srvnode-1/network/mgmt_nw/dns_servers PRVSNR_UNDEFINED
+    else
+        provisioner pillar_set cluster/srvnode-1/network/mgmt_nw/dns_servers \"${srv1_public_dns_servers}\"
```
[the remark this answers is not in the document]
Comment (ypise) [reply cli/src/factory_ops/unboxing/config_update]: Addressed

File: cli/src/factory_ops/unboxing/config_update [hunk 5]
```diff
@@ -335,35 +335,64 @@ function update_pillar_from_local {
...
+        provisioner pillar_set cluster/srvnode-2/network/mgmt_nw/search_domains PRVSNR_UNDEFINED
+    else
+        provisioner pillar_set cluster/srvnode-2/network/mgmt_nw/search_domains \"${srv2_search_domains}\"
+    fi
+
+    srv2_public_dns_servers=$(salt-call --local pillar.get cluster:srvnode-2:network:mgmt_nw:dns_servers --output=newline_values_only)
+    if [[ "${srv2_public_dns_servers}" == "None" ]]; then
+        provisioner pillar_set cluster/srvnode-2/network/mgmt_nw/dns_servers PRVSNR_UNDEFINED
+    else
+        provisioner pillar_set cluster/srvnode-2/network/mgmt_nw/dns_servers \"${srv2_public_dns_servers}\"
```
[the remark this answers is not in the document]
Comment (ypise) [reply cli/src/factory_ops/unboxing/config_update]: Addressed

File: cli/src/factory_ops/unboxing/config_update [hunk 6]
```diff
@@ -335,35 +335,64 @@ function update_pillar_from_local {
...
     else
-        provisioner pillar_set cluster/srvnode-2/network/mgmt_nw/netmask \"${srv1_public_netmask}\"
+        provisioner pillar_set cluster/srvnode-2/network/mgmt_nw/netmask \"${srv2_public_netmask}\"
+    fi
+
+    srv2_search_domains=$(salt-call --local pillar.get cluster:srvnode-2:network:mgmt_nw:search_domains --output=newline_values_only)
+    if [[ "${srv2_search_domains}" == "None" ]]; then
+        provisioner pillar_set cluster/srvnode-2/network/mgmt_nw/search_domains PRVSNR_UNDEFINED
+    else
+        provisioner pillar_set cluster/srvnode-2/network/mgmt_nw/search_domains \"${srv2_search_domains}\"
```
[the remark this answers is not in the document]
Comment (ypise) [reply cli/src/factory_ops/unboxing/config_update]: Addressed

**Comment by ypise** [comment]
@andkononykhin2 
The concerns have been addressed. Consider for another review.

**Review by 83bhp — APPROVED** [review]

**Review by mazinamdar — APPROVED** [review]

**Review by andkononykhin2 — APPROVED** [review]

**Pull request merged**